java/impl/gcp/src/main/java/co/worklytics/psoxy/GcpModule.java

```diff
@@ -79,10 +83,15 @@ static ConfigService nativeConfigService(EnvVarsConfigService envVarsConfigServi
                     // Default is considered as empty; otherwise it will fail due a NPE
                 .orElse("");
 
+
+        SecretManagerConfigService shared = secretManagerConfigServiceFactory.create(ServiceOptions.getDefaultProjectId(), pathToSharedConfig);
+
+        Duration proxyInstanceConfigCacheTtl = Duration.ofMinutes(5);
+        Duration sharedConfigCacheTtl = Duration.ofMinutes(20);
         return CompositeConfigService.builder()
-                .preferred(instanceConfigService)
-                .fallback(secretManagerConfigServiceFactory.create(ServiceOptions.getDefaultProjectId(), pathToSharedConfig))
-                .build();
+            .preferred(new CachingConfigServiceDecorator(instanceConfigService, proxyInstanceConfigCacheTtl))
+            .fallback(new CachingConfigServiceDecorator(shared, sharedConfigCacheTtl))
```

Comment:
GCP wasn't actually using a caching config service, probably bc I expected most stuff we wanted to cache to be bound as env vars (which prior to v0.4.33, they are)